### PR TITLE
Release 24.1.0

### PR DESCRIPTION
**Pull request by jezdez**
## Description

✂️ snip snip ✂️ the making of a new release.

Xref #13486

### CODE REVIEW

File: .authors.yml [hunk 1]
```diff
@@ -2431,3 +2431,13 @@
   github: AniketP04
   num_commits: 1
   first_commit: 2023-10-23 07:46:44
+- name: Schuyler Martin
+  email: schuylermartin45@gmail.com
+  num_commits: 1
+  first_commit: 2023-12-05 10:47:28
+  github: schuylermartin45
+- name: samhaese
```
Comment (kenodegard) [inline .authors.yml:2439]: Got name from signed CLA

```suggestion
- name: Sam Haese
  aliases:
  - samhaese
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,86 @@
...
+
+### Enhancements
+
+* Verify signatures on to-be-installed packages instead of on all packages. (#11545, #13053)
+* Add a new `pre-solves` plugin hook. (#13053)
+* Add a new `post-solves` plugin hook. (#13053)
+* Add support for Python 3.12. (#13072)
+* Check `repodata.json.zst` for faster repodata downloads. (#13256)
+* Add `--skip-flexible-search` option to skip flexible search. (#13315)
+* Adds type hinting for `VersionOrder` class. (#13380)
```
Comment (kenodegard) [inline CHANGELOG.md:13]:
```suggestion
* Add type hinting for `VersionOrder` class. (#13380)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,86 @@
...
+* Provide a more useful warning when attempting to rename a non-existent prefix. (#13387)
+* Add a new flag `--keep-env` to be used with `conda remove --all`. It allows users to delete all packages in the environment while retaining the environment itself. (#13419)
+* Add a Y/N prompt warning users that `conda env remove` and `conda remove --all` deletes not only the conda packages but the entirety of the specified environment. (#12222 via #13440)
+* Add `--no-repodata-zst` flag to skip `repodata.json.zst` check. Useful if the repo returns unparseable `repodata.json.zst` instead of correct data or 404. (#13504)
+
+### Bug fixes
+
+* Create the `~/.conda` directory before trying to write to the `environments.txt` file. (#13338)
+* Ensure PackageRecord.timestamp is dumped in milliseconds. (#13483)
```
Comment (kenodegard) [inline CHANGELOG.md:22]:
```suggestion
* Ensure `PackageRecord.timestamp` is dumped in milliseconds. (#13483)
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,86 @@
...
+* Mark `conda_env/specs/binstar` as pending deprecation. Use `conda.env.specs.binstar` instead. (#13168)
+* Mark `conda_env/specs/requirements` as pending deprecation. Use `conda.env.specs.requirements` instead. (#13168)
+* Mark `conda_env/specs/yaml_file` as pending deprecation. Use `conda.env.specs.yaml_file` instead. (#13168)
+* Mark `conda.testing.integration.make_temp_package_cache` as pending deprecation. (#13511)
+
+### Docs
+
+* Update Getting Started documentation in User Guide. (#13190)
+* Add goat counter (https://www.goatcounter.com/) as an analytics tool. (#13384)
+* Added type hints and doc strings to `conda.cli.main_info`. (#13445)
```
Comment (kenodegard) [inline CHANGELOG.md:52]:
```suggestion
* Add type hints and doc strings to `conda.cli.main_info`. (#13445)
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,86 @@
...
+* Added type hints and doc strings to `conda.cli.main_search`. (#13465)
```
Comment (kenodegard) [inline CHANGELOG.md:53]:
```suggestion
* Add type hints and doc strings to `conda.cli.main_search`. (#13465)
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,86 @@
...
+
+### Other
+
+* Move `conda_env` modules into `conda` while allowing old `conda env` commands to still work via redirect imports. (#13168)
```
Comment (kenodegard) [inline CHANGELOG.md:57]:
```suggestion
* Move `conda_env` modules into `conda` while allowing old `conda env` commands to still work via import redirects. (#13168)
```

**Review by kenodegard — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,86 @@
...
+## 24.1.0 (2024-01-24)
+
+### Enhancements
+
+* Verify signatures on to-be-installed packages instead of on all packages. (#11545, #13053)
+* Add a new `pre-solves` plugin hook. (#13053)
+* Add a new `post-solves` plugin hook. (#13053)
+* Add support for Python 3.12. (#13072)
+* Check `repodata.json.zst` for faster repodata downloads. (#13256)
+* Add `--skip-flexible-search` option to skip flexible search. (#13315)
```
Comment (jaimergp) [inline CHANGELOG.md:12]: This is part of `conda search`, right? Might not be obvious to all readers. What about:

```suggestion
* Add `--skip-flexible-search` option in `conda search` to skip flexible search. (#13315)
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,86 @@
...
+* Add type hinting for `VersionOrder` class. (#13380)
```
Comment (jaimergp) [inline CHANGELOG.md:13]: I don't know if this belongs in Enhancements vs Other.

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,86 @@
...
+* Provide a more useful warning when attempting to rename a non-existent prefix. (#13387)
+* Add a new flag `--keep-env` to be used with `conda remove --all`. It allows users to delete all packages in the environment while retaining the environment itself. (#13419)
+* Add a Y/N prompt warning users that `conda env remove` and `conda remove --all` deletes not only the conda packages but the entirety of the specified environment. (#12222 via #13440)
+* Add `--no-repodata-zst` flag to skip `repodata.json.zst` check. Useful if the repo returns unparseable `repodata.json.zst` instead of correct data or 404. (#13504)
```
Comment (jaimergp) [inline CHANGELOG.md:17]: What's the equivalent option in `.condarc`? Out of curiosity.

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,86 @@
...
+
+### Bug fixes
+
+* Create the `~/.conda` directory before trying to write to the `environments.txt` file. (#13338)
+* Ensure `PackageRecord.timestamp` is dumped in milliseconds. (#13483)
```
Comment (jaimergp) [inline CHANGELOG.md:22]: This is a tests-only PR, IIRC? It didn't have any news because of that.

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,5 +1,86 @@
...
+* Fix error when setting a non-default repodata filename via `CONDA_REPODATA_FNS`. (#13490)
+* Fix the config file location where the integrated Anaconda client gateway loads user configuration from. This is a regression introduced in conda 23.11.0 when the `platformdirs` library was adopted. (#13517)
```
Comment (jaimergp) [inline CHANGELOG.md:24]:
```suggestion
* Fix the config file location where the integrated Anaconda client gateway loads user configuration from. This is a regression introduced in conda 23.11.0 when the `platformdirs` library was adopted. (#13517 via #13520)
```

Aren't we doing `via PR` notes anymroe?

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,5 +1,86 @@
...
+### Docs
+
+* Update Getting Started documentation in User Guide. (#13190)
+* Add goat counter (https://www.goatcounter.com/) as an analytics tool. (#13384)
+* Add type hints and doc strings to `conda.cli.main_info`. (#13445)
+* Add type hints and doc strings to `conda.cli.main_search`. (#13465)
+
+### Other
+
+* Move `conda_env` modules into `conda` while allowing old `conda env` commands to still work via import redirects. (#13168)
```
Comment (jaimergp) [inline CHANGELOG.md:57]: Do you think this should be highlighted? In the top section like we did with conda-libmamba-solver? Maybe along with the new plugin hooks? This changelog ended up being BIG :D

File: CHANGELOG.md (hunk 9 above)
Comment (jezdez) [reply CHANGELOG.md:17]: @dholth Do you know?

File: CHANGELOG.md (hunk 10 above)
Comment (jezdez) [reply CHANGELOG.md:22]: Citing Hyrum's law, somewhere someone is using this testing helper and it is part of the conda package, so official API, and therefore worth mentioning.

File: CHANGELOG.md (hunk 11 above)
Comment (jezdez) [reply CHANGELOG.md:24]: Sometimes yes, sometimes no.

File: CHANGELOG.md (hunk 8 above)
Comment (jezdez) [reply CHANGELOG.md:13]: I'm ambivalent about it, don't you think it's an enhancement? I guess that section is more about enhancements for "end users"?

Comment (beeankha) [reply CHANGELOG.md:13]: That's a good point about enhancements for end users; I vote for this to move to the "Other" section

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,5 +1,86 @@
...
+* Verify signatures on to-be-installed packages instead of on all packages. (#11545, #13053)
+* Add a new `pre-solves` plugin hook. (#13053)
+* Add a new `post-solves` plugin hook. (#13053)
+* Add support for Python 3.12. (#13072)
+* Check `repodata.json.zst` for faster repodata downloads. (#13256)
+* Add `--skip-flexible-search` option in `conda search` to skip flexible search. (#13315)
+* Add type hinting for `VersionOrder` class. (#13380)
+* Provide a more useful warning when attempting to rename a non-existent prefix. (#13387)
+* Add a new flag `--keep-env` to be used with `conda remove --all`. It allows users to delete all packages in the environment while retaining the environment itself. (#13419)
+* Add a Y/N prompt warning users that `conda env remove` and `conda remove --all` deletes not only the conda packages but the entirety of the specified environment. (#12222 via #13440)
```
Comment (beeankha) [inline CHANGELOG.md:16]:
```suggestion
* Add a Y/N prompt warning users that `conda env remove` and `conda remove --all` deletes not only the conda packages but the entirety of the specified environment. (#13440)
```

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,5 +1,86 @@
...
+* Mark `conda_env/specs` as pending deprecation. Use `conda.env.specs` instead. (#13168)
+* Mark `conda_env/specs/binstar` as pending deprecation. Use `conda.env.specs.binstar` instead. (#13168)
+* Mark `conda_env/specs/requirements` as pending deprecation. Use `conda.env.specs.requirements` instead. (#13168)
+* Mark `conda_env/specs/yaml_file` as pending deprecation. Use `conda.env.specs.yaml_file` instead. (#13168)
+* Mark `conda.testing.integration.make_temp_package_cache` as pending deprecation. (#13511)
+
+### Docs
+
+* Update Getting Started documentation in User Guide. (#13190)
+* Add goat counter (https://www.goatcounter.com/) as an analytics tool. (#13384)
```
Comment (beeankha) [inline CHANGELOG.md:51]:
```suggestion
* Add GoatCounter (https://www.goatcounter.com/) as an analytics tool. (#13384)
```

File: CHANGELOG.md (hunk 9 above)
Comment (dholth) [reply CHANGELOG.md:24]: It looks like we need to add the condarc option which would have to be `no_repodata_zst: true`

I also noticed a `repodata_use_zst: true` in my condarc which is a mamba option.

Comment (jezdez) [reply CHANGELOG.md:24]: Sounds like something to fix in 24.1.1? Or blocking the release?

Comment (dholth) [reply CHANGELOG.md:24]: https://github.com/conda/conda/issues/13528

Comment (jezdez) [reply CHANGELOG.md:24]: Thank you @dholth

**Review by jaimergp — APPROVED** [review]

File: CHANGELOG.md (hunk 9 above)
Comment (dholth) [reply CHANGELOG.md:24]: Oh, the environment variable works in the test suite https://github.com/conda/conda/blob/main/tests/gateways/test_jlap.py#L371-L378 but for some reason it isn't working from the command line.

Comment (dholth) [reply CHANGELOG.md:24]: Yes, that was the issue. See https://github.com/conda/conda/pull/13530

**Pull request merged**